core-primitives/attestation-handler/src/attestation_handler.rs

```diff
@@ -119,83 +119,37 @@ where
 	OCallApi: EnclaveAttestationOCallApi,
 	NodeMetadataRepository: AccessNodeMetadata<MetadataType = NodeMetadata>,
 {
-	fn perform_ra(
+	fn perform_ra<ExtrinsicsFactory: CreateExtrinsics>(
 		&self,
-		genesis_hash_slice: &[u8],
-		nonce: u32,
-		url_slice: &[u8],
+		url: String,
+		extrinsics_factory: Arc<ExtrinsicsFactory>,
```

Comment:
The extrinsics factory should be a dependency of the attestation handler (member, set on construction), and not be passed in when calling this function.

This means that the attestation handler component needs to be initialized after the extrinsics factory, so the ordering needs to change.

Reply:
Is it? That would make the attestation a parentchain specific component. Should that really be the case?

Reply:
It has that dependency, yes. What I'm proposing is that the dependency should not be part of the interface, just the implementation. It does not change the fact that we have that dependency.

Reply:
Maybe we should think about removing the dependency, by changing the interface. E.g. that the attestation handler just returns the certificate, but is not responsible packing and sending the extrinsic to a chain.

Reply:
That way, whatever parentchain component needs to do remote attestation, can use the attestation handler to create that report/certificate and then send it to its own parentchain.

Reply:
Jup jup, I like that. Will try :)